### PR TITLE
Fix Trace List with variable

### LINKED ISSUE #131: x-ray data source variable binding not working for Trace List query parameter
**What happened**:
grafana-x-ray-datasource

Grafana x-ray data source does not behave correctly when selecting Trace List and using a dashboard variable. It does not filter the trace list.

If we directly put the traceId it works fine. It seems variable interpolation is causing a different AWS query to be called (`getTraceSummary`) as described in the screen recording and below screenshot

image

**What you expected to happen**:
Grafana x-ray data source should have used the dashboard variable to query a single trace and render the graph.

**How to reproduce it (as minimally and precisely as possible)**:
- Create a x-ray data source
- Query a x-ray trace
**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.4.7
- Data source type & version: x-ray , 1.4.0
- OS Grafana is installed on: Amazon Managed Grafana
- User OS & Browser: Windows, Chrome
- Grafana plugins: x-ray
- Others:

### PR DESCRIPTION
**Pull request by iwysiu**
The query editor uses a check for `trace id` to determine if a `Trace List` query is a `getTrace` or `getTraceSummaries` query. Previously, it did not replace variables before checking, which meant that a variable that represented a trace id would incorrectly get sorted as a `getTraceSummaries` query.

Structure-wise it would probably be better if the query function determined which type it was, but I wasn't sure we wanted to do that much restructuring.

Fixes #131

### CODE REVIEW

**Review by fridgepoet — APPROVED** [review]
LGTM but would prefer someone else approve as well

File: src/components/QueryEditor/QueryEditor.test.tsx [hunk 1]
```diff
@@ -3,7 +3,47 @@ import { render, screen, fireEvent, act, waitFor } from '@testing-library/react'
 import { QueryEditor } from './QueryEditor';
 import { Group, Region, XrayJsonData, XrayQuery, XrayQueryType } from '../../types';
 import { XrayDataSource } from '../../DataSource';
-import { DataSourceInstanceSettings } from '@grafana/data';
+import { DataSourceInstanceSettings, ScopedVars, VariableModel } from '@grafana/data';
+import { TemplateSrv } from '@grafana/runtime';
+
+jest.mock('@grafana/runtime', () => {
```
Comment (sarahzinger) [inline src/components/QueryEditor/QueryEditor.test.tsx:9]: it seems like you're updating a mock but not any tests in this file. Is that intentional?

File: src/components/QueryEditor/QueryEditorForm.test.tsx [hunk 2]
```diff
@@ -1,10 +1,61 @@
 import { queryTypeOptionToQueryType } from './QueryEditorForm';
 import { queryTypeOptions } from './constants';
 import { XrayQueryType } from '../../types';
+import { TemplateSrv } from '@grafana/runtime';
+import { ScopedVars, VariableModel } from '@grafana/data';
+
+jest.mock('@grafana/runtime', () => {
```
Comment (sarahzinger) [inline src/components/QueryEditor/QueryEditorForm.test.tsx:7]: is there a way to just mockout getTemplateSrv ? It seems like that's the only bit you're using right?

**Review by sarahzinger — COMMENTED** [review]
Looking good just a few clarifying questions first if that's cool!

File: src/components/QueryEditor/QueryEditorForm.test.tsx (hunk 2 above)
Comment (iwysiu) [reply src/components/QueryEditor/QueryEditorForm.test.tsx]: switched it to a jest.SpyOn

**Pull request merged**